integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/log/LogMediatorLevelsAndCategoryTestCase.java

```diff
@@ -0,0 +1,183 @@
+import org.testng.annotations.Test;
+import org.wso2.carbon.integration.common.admin.client.LogViewerClient;
+import org.wso2.carbon.logging.view.stub.types.carbon.LogEvent;
+import org.wso2.esb.integration.common.clients.logging.LoggingAdminClient;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+
+/**
+ * THis test case verifies different levels and categories in log mediator
+ */
+public class LogMediatorLevelsAndCategoryTestCase extends ESBIntegrationTest {
+
+    private LogViewerClient logViewer;
+    private LoggingAdminClient logAdmin;
+    private OMElement response;
+    private LogEvent[] logs;
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        super.init();
+        logAdmin = new LoggingAdminClient(contextUrls.getBackEndUrl(), getSessionCookie());
+        logViewer = new LogViewerClient(contextUrls.getBackEndUrl(), getSessionCookie());
+        logViewer.clearLogs();
+        verifyProxyServiceExistence("LogMediatorLevelAndCategoryTestProxy");
+        logAdmin.updateLoggerData("org.apache.synapse.mediators.builtin",
+                LoggingAdminClient.LogLevel.TRACE.name(),true, false);
+        response = axis2Client
+                .sendSimpleStockQuoteRequest(getProxyServiceURLHttp("LogMediatorLevelAndCategoryTestProxy"),
+                        null, "WSO2");
+        Assert.assertTrue(response.toString().contains("WSO2"));
+        logs = logViewer.getAllRemoteSystemLogs();
+    }
+
+    @Test(groups = "wso2.esb", description = "Test debug level log")
+    public void testDebugLevelLogs() throws Exception {
+        Boolean isDebugLogAvailable = false;
+        for (LogEvent logEvent : logs) {
+            String message = logEvent.getMessage();
+            if (message.contains("*****LOGGING IN DEBUG CATEGORY - CUSTOM LEVEL*****")) {
+                isDebugLogAvailable = true;
+            }
+        }
+        Assert.assertTrue(isDebugLogAvailable, "DEBUG Log not found. Debug logs not working properly");
+    }
+
+    //This test case disabled due to issue reported as "[EI 620][Integration Profile] Trace logs not going to indicate
+    // via UI" (https://github.com/wso2/product-ei/issues/1071)
+    @Test(groups = "wso2.esb", description = "Test trace level log", enabled = false)
+    public void testTraceLevelLogs() throws Exception {
+        Boolean isTraceLogAvailable = false;
+        for (LogEvent logEvent : logs) {
+            String message = logEvent.getMessage();
+            if (message.contains("*****LOGGING IN TRACE CATEGORY - CUSTOM LEVEL*****")) {
+                isTraceLogAvailable = true;
+            }
+        }
+        Assert.assertTrue(isTraceLogAvailable, "TRACE Log not found. Trace logs not working properly");
+    }
+
+    @Test(groups = "wso2.esb", description = "Test info level log")
+    public void testInfoLevelLogs() throws Exception {
+        Boolean isInfoLogAvailable = false;
+        for (LogEvent logEvent : logs) {
```

Comment:
Can't we move this 'for' loop to a common method? It seems same functionality is getting repeated.

Reply:
Yes, Will do.